src/main/kotlin/cacophony/controlflow/functions/FunctionHandlerImpl.kt

```diff
@@ -93,31 +103,23 @@ class FunctionHandlerImpl(
             callerFunction.generateAccessToFramePointer(ancestorFunctionHandlers.first().getFunctionDeclaration())
         }
 
-    override fun generateVariableAccess(variable: Variable): CFGNode.LValue {
-        val definedInDeclaration =
-            when (variable) { // TODO: remove when on variable type
-                is Variable.SourceVariable -> {
-                    // TODO: adjust to new analyzedFunction semantics
-                    analyzedFunction.variables.find { it.declaration == variable.definition }?.definedIn
-                }
-                is Variable.AuxVariable.StaticLinkVariable -> {
-                    if (getStaticLink() == variable) {
-                        function
-                    } else {
-                        ancestorFunctionHandlers.find { it.getStaticLink() == variable }?.getFunctionDeclaration()
-                    }
-                }
-                else -> throw GenerateVariableAccessException(
-                    "Cannot generate access to variables other than static links and source variables.",
-                )
+    override fun generateVariableAccess(variable: Variable.PrimitiveVariable): CFGNode.LValue {
+        if (variable in variableAllocation) { // WARN: hack to handle arguments not used inside function
```

Comment:
Problem is when we have an Argument which is not used inside the body of the function. Some functionalities was to iterate over all of arguments and do something with that (e.g. prologue generation), and old generateVariableAccess trhows error it this example

`let f = [x:Int] -> Int => 2`

Would throw error in some places.

Reply:
OK, I see
so I think it should be changed in FunctionAnalysis so that either there is some field `AnalyzedFunction.arguments` or arguments are added to `AnalyzedFunction.variables` even if unused. (we can make our compiler smart and not generate unused function arguments, but I think that would generate more bugs than benefits)
Which version do you prefer? I will make appropriate issue

Reply:
I meean, it can stay with the additional check that I added, but the preferable solution would be to add them to the `AnalyzedFunction.variables` instead of introducing new place where we have to look for the variables in the function.